src/main/java/io/vertx/core/net/impl/pool/CombinerExecutor.java

```diff
@@ -27,7 +28,12 @@ public class CombinerExecutor<S> implements Executor<S> {
   private final Queue<Action<S>> q = PlatformDependent.newMpscQueue();
   private final AtomicInteger s = new AtomicInteger();
   private final S state;
-  private final ThreadLocal<Task> current = new ThreadLocal<>();
+
+  protected final class InProgressTail {
```

Comment:
Can it be `static`?

Reply:
Uh!! Missed that you are right! Please open a pr, we love PRs

Reply:
And many thanks to lookin at it!

Reply:
Glad that help, I'm using Vertx at work too, very great project!